lib/server/plugins/api/index.js

```diff
@@ -118,6 +118,25 @@ exports.register = function (plugin, options, next) {
     );
   };
 
+  // allow clients to request a gzip response, even if the
+  // Accept-Encoding headers is missing or mangled due to
+  // faulty proxy servers
+  // http://www.stevesouders.com/blog/2010/07/12/velocity-forcing-gzip-compression/
+  plugin.ext('onPreHandler', function maybeForceGzip (request, reply) {
+    var hasGzipQuery = function () {
```

Comment:
We don’t, technically.

The original line looked like this:

> if (request.query.gzip === 'force')

Then I wanted to add the cookie bit:

> if (request.query.gzip === 'force' || request.state.hoodie_force_gzip === 'true')

Which got a little long for my taste, and instead of reformatting or using temporary variables for the `query` and `state` values, I opted for descriptive names, which has the added benefit, that the `if` now reads very naturally:

> if (hasGzipQuery() || hasGzipCookie()) 

We can always refactor this, should it become a problem.